### DISCUSSION TITLE
Should `rake test` be passing on main?

### DISCUSSION THREAD

**Discussion by martinemde** [ruby/prism#1871] [Q&A] [answered]
I pulled down the project, compiled and ran the test suite:

```
2476 tests, 1111929 assertions, 56 failures, 0 errors, 0 pendings, 0 omissions, 0 notifications
```

Is this expected or do I need to solve something in my local environment?

Edit: More info:
1. I ran the test suite once with `rake test`. It failed a number of tests (e.g., `/Users/martinemde/Projects/github.com/ruby/prism/test/prism/parse_test.rb:143`).
2. Then I recompiled with `rake compile`
3. I ran the test suite once again with `rake test` and it failed again on the same specs.
4. Then I filed this question, two failures in a row before and after compiling.
5. I ran the test suite one more time, this time it passed fully.
6. I was confused.

**Comment by martinemde** [comment, marked as answer]
I may have my own answer. Strangely after running it a few times, it solved itself.

Edit: I had made a guess here about why it happened, but that was wrong. I'm not sure what caused the failures twice in a row and then resolve without any changes on my part.

Edit2: I made a fresh clone of the repository, compiled and ran the tests and they all passed. I'm going to mark this as answered since it seems like something transient.

**Comment by kddnewton** [comment]
Hmm, I'm curious what happened, but glad you got it resolved.

**Comment by enebo** [comment]
I have had a failing test suite with a similar number of failures when I accidentally use Ruby 3.1 to build.
